tests/util/mod.rs

```diff
@@ -56,6 +56,32 @@ macro_rules! assert_err {
     }};
 }
 
+/// Expect an event that may not be represented by an Interest, such as
+/// `is_read_closed` and `is_write_closed` on some platforms.
+#[macro_export]
+macro_rules! expect_flaky_event {
```

Comment:
We're really abusing `Interests` in `ExpectEvent`, so in the future we should merge this with `expect_event` and just make a readiness type for testing.